packages/widgets/src/docker/component.tsx

```diff
@@ -0,0 +1,311 @@
+          >
+            <Text size="6cqmin">{(safeValue(row.original.memoryUsage) / (1024 * 1024)).toFixed(2)} MiB</Text>
+          </Badge>
+        );
+      },
+    },
+    {
+      accessorKey: "actions",
+      header: t("action.title"),
+      mantineTableHeadCellProps: {
+        style: {
+          fontSize: "7cqmin",
+          padding: "2cqmin",
+          width: "10%",
+        },
+      },
+      Cell({ row }) {
+        const utils = clientApi.useUtils();
+        const { mutateAsync: startContainer } = clientApi.docker.startAll.useMutation();
+        const { mutateAsync: stopContainer } = clientApi.docker.stopAll.useMutation();
+        const { mutateAsync: restartContainer } = clientApi.docker.restartAll.useMutation();
+
+        const handleActionAsync = async (action: "start" | "stop" | "restart") => {
+          const mutation = action === "start" ? startContainer : action === "stop" ? stopContainer : restartContainer;
+
+          await mutation(
+            { ids: [row.original.id] },
+            {
+              async onSettled() {
+                await utils.docker.getContainers.invalidate();
+              },
+              onSuccess() {
+                showSuccessNotification({
+                  title: t(`action.${action}.notification.success.title`),
+                  message: t(`action.${action}.notification.success.message`),
+                });
+              },
+              onError() {
+                showErrorNotification({
+                  title: t(`action.${action}.notification.error.title`),
+                  message: t(`action.${action}.notification.error.message`),
+                });
+              },
+            },
+          );
+        };
+
+        return (
+          <Group wrap="nowrap" gap="xs">
+            <Tooltip label={row.original.state === "running" ? t("action.stop.label") : t("action.start.label")}>
+              <ActionIcon
+                variant="transparent"
+                radius={999}
+                size="var(--button-size)"
+                onClick={() => handleActionAsync(row.original.state === "running" ? "stop" : "start")}
+              >
+                {row.original.state === "running" ? (
+                  <IconPlayerStop style={actionIconIconStyle} />
+                ) : (
+                  <IconPlayerPlay style={actionIconIconStyle} />
+                )}
+              </ActionIcon>
+            </Tooltip>
+            <Tooltip label={t("action.restart.label")}>
+              <ActionIcon
+                variant="transparent"
+                radius={999}
+                size="var(--button-size)"
+                onClick={() => handleActionAsync("restart")}
+              >
+                <IconRotateClockwise style={actionIconIconStyle} />
+              </ActionIcon>
+            </Tooltip>
+          </Group>
+        );
+      },
+    },
+  ];
+
+  const { data } = clientApi.docker.getContainersWidget.useQuery();
```

Comment:
Please use `useSuspenseQuery` to automatically show a loading spinner until it has loaded